src/transactions/tests_helpers.rs

```diff
@@ -617,6 +617,82 @@ pub fn derive_transactions(
     unvault_tx.finalize(&secp)?;
     roundtrip!(unvault_tx, UnvaultTransaction);
 
+    // Create a CPFP transaction for the unvault
+    // Some fake listunspent outputs
+    let listunspent = vec![
+        CpfpTxIn::new(
+            OutPoint::from_str(
+                "f21596dd9df36b86bcf65f0884f1f20675c1fc185bc78a37a9cddb4ae5e3dd9f:0",
+            )
+            .unwrap(),
+            CpfpTxOut::new(Amount::from_sat(30_000), &der_cpfp_descriptor),
+        ),
+        CpfpTxIn::new(
+            OutPoint::from_str(
+                "f21596dd9df36b86bcf65f0884f1f20675c1fc185bc78a37a9cddb4ae5e3dd9f:1",
+            )
+            .unwrap(),
+            CpfpTxOut::new(Amount::from_sat(30_000), &der_cpfp_descriptor),
+        ),
+    ];
+
+    // Let's ask for a decent feerate
+    let added_feerate = 6121;
+    let mut cpfp_tx = unvault_tx
+        .clone()
+        .cpfp_transaction(&der_cpfp_descriptor, added_feerate, listunspent.clone())
+        .unwrap();
+    let cpfp_txin = unvault_tx.cpfp_txin(&der_cpfp_descriptor).unwrap();
+
+    // The cpfp tx contains the input of the tx to be cpfped, right?
+    assert!(cpfp_tx.tx().input.contains(&cpfp_txin.unsigned_txin()));
+
+    for o in &cpfp_tx.tx().output {
+        // Either the change is 0 with an OP_RETURN,
+        // or its value is bigger than CPFP_MIN_CHANGE, and we send
+        // back to the cpfp_txin script_pubkey
+        assert!(
+            (o.value == 0 && o.script_pubkey.is_op_return())
+                || (o.value >= CPFP_MIN_CHANGE
+                    && o.script_pubkey == cpfp_txin.txout().txout().script_pubkey)
+        );
+    }
+
+    // We sign and check the package feerate
+    let inputs_len = cpfp_tx.psbt().inputs.len();
+    for i in 0..inputs_len {
+        let cpfp_tx_sighash = cpfp_tx
+            .signature_hash(i, SigHashType::All)
+            .expect("Input exists");
+        satisfy_transaction_input(
+            &secp,
+            &mut cpfp_tx,
+            i,
+            &cpfp_tx_sighash,
+            &mancpfp_priv,
+            Some(child_number),
+        )?;
+    }
+
+    roundtrip!(cpfp_tx, CpfpTransaction);
+    cpfp_tx.finalize(&secp)?;
+    roundtrip!(cpfp_tx, CpfpTransaction);
+
+    // Number of inputs * cpfp value - outputs
+    let cpfp_tx_fees = inputs_len as u64 * 30_000
+        - cpfp_tx
+            .psbt()
+            .global
+            .unsigned_tx
+            .output
+            .iter()
+            .fold(0, |sum, o| sum + o.value);
+    assert!(
+        1000 * (cpfp_tx_fees + unvault_tx.fees())
+            / (cpfp_tx.tx().get_weight() + unvault_tx.tx().get_weight()) as u64
```

Comment:
This is incorrect since you don't consider the satisfaction weight. You need to extract the inner tx for the CPFP and to use `max_weight()` for the Unvault.

This actually makes the test fail. (the weight increases from `924` to `1252` at what i think is the first iteration.)